compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -1251,6 +1251,8 @@ pub enum NonMutatingUseContext {
     UniqueBorrow,
     /// AddressOf for *const pointer.
     AddressOf,
+    /// PlaceMention statement.
```

Comment:
Might be worth adding a clarifying comment on why this is here, given that it used to be in the wrong category.